crates/polars-ops/src/frame/join/hash_join/single_keys.rs

```diff
@@ -1,53 +1,146 @@
+use polars_utils::sync::SyncPtr;
+
 use super::*;
 
+// FIXME: we should compute the number of threads / partition size we'll use.
+// let avail_threads = POOL.current_num_threads();
+// let n_threads = (num_keys / MIN_ELEMS_PER_THREAD).clamp(1, avail_threads);
+const MIN_ELEMS_PER_THREAD: usize = 128;
```

Comment:
Maybe we should set this very low for debug builds so that our tests will hit the interesting code.

Reply:
It's not a bad idea, my only worry is that now you have different behavior in debug vs release which can lead to *other* annoying to debug things.